pkg/services/ngalert/migration/alert_rule_graphite_test.go

```diff
@@ -0,0 +1,88 @@
+package migration
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/grafana/grafana/pkg/components/simplejson"
+	"github.com/grafana/grafana/pkg/infra/log/logtest"
+	"github.com/grafana/grafana/pkg/services/dashboards"
+	"github.com/grafana/grafana/pkg/services/ngalert/models"
+)
+
+func TestMigrateAlertRuleGraphiteQueries(t *testing.T) {
+	tc := []struct {
+		name      string
+		input     *simplejson.Json
+		dashboard *dashboards.Dashboard
+		panelID   int64
+		expected  string
+	}{
+		{
+			name: "when a query has a sub query - it is not unwrapped if unavailable through dasboards",
+			input: simplejson.NewFromAny(map[string]any{
+				"refId":  "C",
+				"target": "scale(asPercent(diffSeries(#B, #A), #B), 100)",
+			}),
+			expected: `{"refId":"C", "target":"scale(asPercent(diffSeries(second.query.count, first.query.count), second.query.count), 100)"}`,
+			panelID:  123,
+			dashboard: &dashboards.Dashboard{
+				Data: simplejson.NewFromAny(map[string]any{
+					"panels": []map[string]any{
+						{
+							"id": 123,
+							"targets": []target{
+								{
+									RefID:  "A",
```

Comment:
Can we add some tests for when some targets are missing? Like `A` is missing for `scale(asPercent(diffSeries(#B, #A), #B), 100)`?